File: app/lib/models/progress_metric.dart
class ProgressMetric {
  final int id;
  final int studentId;
  final int? exerciseId;
  final String? label;
  final String metricType;
  final double value;
  final String? unit;
  final String recordedAt;

  ProgressMetric({
    required this.id,
    required this.studentId,
    this.exerciseId,
    this.label,
    required this.metricType,
    required this.value,
    this.unit,
    required this.recordedAt,
  });

  factory ProgressMetric.fromJson(Map<String, dynamic> json) => ProgressMetric(
        id: json['id'],
        studentId: json['student_id'],
        exerciseId: json['exercise_id'],
        label: json['label'],
        metricType: json['metric_type'],
        value: double.tryParse(json['value'].toString()) ?? 0,
        unit: json['unit'],
        recordedAt: json['recorded_at'],
      );
}

File: backend/app/Models/ProgressMetric.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ProgressMetric extends Model
{
    protected $fillable = ['student_id', 'exercise_id', 'label', 'metric_type', 'value', 'unit', 'recorded_at'];

    protected $casts = ['recorded_at' => 'date'];

    public function student(): BelongsTo
    {
        return $this->belongsTo(Student::class);
    }

    public function exercise(): BelongsTo
    {
        return $this->belongsTo(Exercise::class);
    }
}
